app/src/org/commcare/tasks/ResourceEngineTask.java

```diff
@@ -129,7 +126,7 @@ public void compoundResourceAdded(final ResourceTable table) {
         synchronized (statusLock) {
             // if last time isn't set or is less than our spacing count, do not
             // perform status update. Also if we are already running one, just skip this.
-            if (statusCheckRunning || System.currentTimeMillis() - lastTime < ResourceEngineTask.STATUS_UPDATE_WAIT_TIME) {

```

Comment:
I'm a bit confused how this could have been causing it to get stuck, would you mind explaining?

Also, should delete the `STATUS_UPDATE_WAIT_TIME` variable, which is now unused.